cardano-api/src/Cardano/Api/IO.hs

```diff
@@ -99,15 +133,9 @@ handleFileForWritingWithOwnerPermission path f = do
     (targetDir, targetFile) = splitFileName path
 #endif
 
-newtype OutputFile = OutputFile
-  { unOutputFile :: FilePath
-  }
-  deriving Generic
-  deriving newtype (Eq, Ord, Show, IsString, ToJSON, FromJSON)
-
-writeByteStringFile :: MonadIO m => FilePath -> ByteString -> m (Either (FileError ()) ())
+writeByteStringFile :: MonadIO m => File 'Out -> ByteString -> m (Either (FileError ()) ())
 writeByteStringFile fp bs = runExceptT $
```

Comment:
All of these functions now take the new file type as the argument.  This reduces the opportunity for introducing bugs for instance writing to a file that was annotated as `In` or reading from a file that was annotated as `Out`.